Task2

Saxophilyman opened the pull request.

Saxophilyman committed:
Part_II "After HOLLIDAYS FuckSelfStudies" (1.2)
- ctrl+alt+l;

Themes for working:
- negative sign in expression;

Saxophilyman committed:
Part_III "From PullRequests 1.1"
 - points 1, 7, 9.
-------------------------------

Saxophilyman committed:
Part_III "From PullRequests 1.2"
 - correction

Saxophilyman committed:
Part_IV "Calculator 4.0 "
 (new functional - memory)

 added:
 - ArrayList<String> listOfLastOperations;
 - isUserRequestsMemory();
 - printLastOperations();
 - addOperationToListOfLastOperations();
 - correcting startWorkingCalculator();
   ----------------------

 for working:
 - DATE;
 - use addOperationToListOfLastOperations();

Saxophilyman committed:
Part_IV "Calculator 4.1 "
 (new functional - memory)
 added:
 - corrected text in Main;
 - added README.md

 for working:
 - DATE;
 - use addOperationToListOfLastOperations();

Saxophilyman committed:
Part_IV "Calculator 4.1 "
 (new functional - memory)

 working:
 - use addOperationToListOfLastOperations();

 -------------------

 for future working:
 - DATE;

Saxophilyman committed:
Part_IV "Calculator 4.2"
 (new functional - memory)

 add:
 - concatenationInformationOfOperation();
 - private BigDecimal result;
 - apply addOperationToListOfLastOperations();

 -------------------

 working:
 - use DATE in concatenationInformationOfOperation();

Saxophilyman committed:
Part_IV "Calculator 4.2 "
 (new functional - memory)

 add:
 - timeOfOperation() using GregorianCalendar()+SimpleDateFormat()

Saxophilyman committed:
Part_IV "Calculator 4.3"
 (new functional - memory)

 add:
 - timeOfOperation() using GregorianCalendar()+SimpleDateFormat()
 - ctrl+alt+l

Saxophilyman committed:
Part_IV "Calculator 4.4"
 (new functional - memory)

 add:
 - check yellow marks;

epereskokov commented on src/Calculator.java:
```diff
@@ -1,72 +1,115 @@
+    private boolean isUserRequestsMemory(String isReadConsoleTextMemory) {
+        if (isReadConsoleTextMemory.equals("memory")) {
+            /*???*/
+            return printLastOperations(listOfLastOperations);
```
printLastOperations(listOfLastOperations);
return true;

epereskokov commented on src/Calculator.java:
```diff
@@ -1,72 +1,115 @@
 
 
-    private static void checkReadConsoleIsNumber(String readConsoleText) {
+    //нихуя не понятны приколы с булеаном и ретёрном
```
что за приколы?
//та же return value of the method is never used - я не вижу в иде этого ворнинга, все ок.

epereskokov commented on src/Calculator.java:
```diff
@@ -1,72 +1,115 @@
+    }
+
+    private void addOperationToListOfLastOperations(ArrayList<String> listOfLastOperations, String Operation) {
+        if (listOfLastOperations.size() >= 10) {
```
magicNumbers, числа должны храниться в static final константах с читаемыми названиями.

epereskokov commented on src/Calculator.java:
```diff
@@ -1,72 +1,115 @@
+
+    private void addOperationToListOfLastOperations(ArrayList<String> listOfLastOperations, String Operation) {
+        if (listOfLastOperations.size() >= 10) {
+            listOfLastOperations.remove(0);
```
В целом все правильно. Но я хочу, чтобы ты выделил работу с памятью в отдельный класс OperationsRepository (у нас же ООП) и взаимодействовал с инстансом этого отдельного класса.

Saxophilyman committed:
Part_IV "Calculator 4.5"
 (new functional - memory)

Corrections:
 - Magick Numbers;

Saxophilyman committed:
Part_IV "Calculator 5.0"
 (try memory in OOP)

Saxophilyman committed:
Part_IV "Calculator 5.0"
 (try memory in OOP)

epereskokov commented on src/OperationsRepository.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+
+    public void addOperationToListOfLastOperations(ArrayList<String> listOfLastOperations, String Operation) {
```
у тебя суть репозитория в том, что он обеспечивает хранение данных и операции с ним. Соответственно этот List должен храниится внутри этого класса, а не снаружи.

epereskokov commented on src/OperationsRepository.java:
```diff
@@ -0,0 +1,47 @@
+//        return false;
+//    }
+//
+    public boolean printLastOperations(ArrayList<String> listOfLastOperations) {
```
репозиторий хранит и обеспечивает доступ к данным, он сам ничего не должен печатать.
По сути снаружи он должен выглядеть как объект с методами addNewOperation(..) и getAllOperations()

Saxophilyman replied:
так, непонел, а что за метод getAllOperations()? откуда он и куда?

epereskokov commented on src/Calculator.java:
```diff
@@ -77,148 +121,152 @@ private static void checkReadConsoleIsNumber(String readConsoleText) {
+        return symbolOfReadConsole.equals(".");
+    }
+
+    static String timeOfOperation() {
```
глагол в названии имени метода где?
DateFormat timeOfOperation можно сохранить в переменную, чтобы каждый раз ее не создавать.
Вместо  Calendar calendar = new GregorianCalendar().getTime() можно еще использовать System.currentTimeMillis(), это чисто к сведению.

Saxophilyman committed:
Part_IV "Calculator 5.1"
 (try memory in OOP)

Saxophilyman committed:
Part_IV "Calculator 5.2"
 (try memory in OOP)

 add:
 - get getListOfLastOperations();
 - change concatenationInformationOfOperation;
 - deleted comments;

Saxophilyman committed:
Part_IV "Calculator 5.3"
 (try memory in OOP)

 -corrections;

Saxophilyman committed:
Part_IV "Calculator 5.3"
 (try memory in OOP)

 -corrections;
